NDTensors/src/lib/Sectors/src/abstractcategory.jl

```diff
@@ -1,45 +1,129 @@
+# This file defines the abstract type AbstractCategory
+# all fusion categories (Z{2}, SU2, Ising...) are subtypes of AbstractCategory
+
 abstract type AbstractCategory end
 
-label(c::AbstractCategory) = error("method `label` not defined for type $(typeof(c))")
+# ===================================  Base interface  =====================================
+function Base.isless(c1::C, c2::C) where {C<:AbstractCategory}
+  return isless(category_label(c1), category_label(c2))
+end
 
-function dimension(c::AbstractCategory)
-  return error("method `dimension` not defined for type $(typeof(c))")
+# =================================  Sectors interface  ====================================
+trivial(x) = trivial(typeof(x))
+function trivial(axis_type::Type{<:AbstractUnitRange})
+  return GradedAxes.gradedrange([trivial(eltype(axis_type))])  # always returns nondual
+end
+function trivial(la_type::Type{<:LabelledNumbers.LabelledInteger})
+  return la_type(1, trivial(LabelledNumbers.label_type(la_type)))
+end
+function trivial(type::Type)
+  return error("`trivial` not defined for type $(type).")
 end
 
-function label_fusion_rule(category_type::Type{<:AbstractCategory}, l1, l2)
-  return error("`label_fusion_rule` not defined for type $(category_type).")
+istrivial(c::AbstractCategory) = (c == trivial(c))
+
+function category_label(c::AbstractCategory)
+  return error("method `category_label` not defined for type $(typeof(c))")
 end
 
-function fusion_rule(c1::AbstractCategory, c2::AbstractCategory)
-  category_type = typeof(c1)
-  return [category_type(l) for l in label_fusion_rule(category_type, label(c1), label(c2))]
+block_dimensions(g::AbstractUnitRange) = block_dimensions(SymmetryStyle(g), g)
+block_dimensions(::AbelianGroup, g) = GradedAxes.unlabel.(BlockArrays.blocklengths(g))
```

Comment:
```suggestion
block_dimensions(::AbelianGroup, g) = LabelledNumbers.unlabel.(BlockArrays.blocklengths(g))
```
to be more specific (unless there is a special version of `GradedAxes.unlabel` you are intending to use, but I don't see why there would be).